src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/shared/GenericString.java

```diff
@@ -0,0 +1,59 @@
+package ay2122s1_cs2103t_w16_2.btbb.model.shared;
+
+import static ay2122s1_cs2103t_w16_2.btbb.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Generic String in the address book, that is reusable.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGenericString(String)}
+ */
+public class GenericString {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Generic Strings should only contain alphanumeric characters and spaces, and it should not be blank";
```

Comment:
It doesn't make sense to show users the term "Generic Strings" can we have classes that extend from Generic String?

E.g. 
```
public class Unit extends GenericString {
    public static final String MESSAGE_CONSTRAINTS =
            "Units should only contain alphanumeric characters and spaces, and it should not be blank";

    /**
     * Constructs a {@code GenericString}.
     *
     * @param genericString A valid generic string.
     */
    public Unit(String genericString) {
        super(genericString);
    }
}
```

Sorry for the trouble.

The other way is to use a `getMessageConstraints(String attributeName)` method.

What do you think is better?

Reply:
going to do getMessageConstraints instead, due to time constraint.